src/llama-model.cpp

```diff
@@ -469,6 +470,10 @@ void llama_model::load_hparams(llama_model_loader & ml) {
     std::fill(hparams.n_head_arr.begin(),    hparams.n_head_arr.end(),    0);
     std::fill(hparams.n_head_kv_arr.begin(), hparams.n_head_kv_arr.end(), 0);
     std::fill(hparams.n_ff_arr.begin(),      hparams.n_ff_arr.end(),      0);
+    std::fill(
```

Comment:
On this branch, this is somewhat incomplete. For the Granite 4 branch, I have this array further overridden by parsing `LLM_KV_ATTENTION_LAYER_INDICES`. I'm wondering if that logic should be universal enough that I should hoist it to this portion here to be done in a model-agnostic way.